erpnext/controllers/buying_controller.py

```diff
@@ -174,6 +174,7 @@ def validate_for_subcontracting(self):
 					item.bom = None
 
 	def create_raw_materials_supplied(self, raw_material_table):
+		self.set(raw_material_table, [])
```

Comment:
Don't make it blank every time. Because user enters serial nos and batch nos in the raw material table, those will get blank.